rasa/telemetry.py

```diff
@@ -628,50 +645,249 @@ async def track_model_training(
+            "training_id": training_id,
+            "type": model_type,
+            "pipeline": config.get("pipeline"),
+            "policies": config.get("policies"),
+            "num_intent_examples": len(nlu_data.intent_examples),
+            "num_entity_examples": len(nlu_data.entity_examples),
+            "num_actions": len(domain.action_names),
+            # Old nomenclature from when 'responses' were still called
+            # 'templates' in the domain
+            "num_templates": len(domain.templates),
+            "num_slots": len(domain.slots),
+            "num_forms": len(domain.forms),
+            "num_intents": len(domain.intents),
+            "num_entities": len(domain.entities),
+            "num_story_steps": len(stories.story_steps),
+            "num_lookup_tables": len(nlu_data.lookup_tables),
+            "num_synonyms": len(nlu_data.entity_synonyms),
+            "num_regexes": len(nlu_data.regex_features),
+        },
     )
     start = datetime.now()
     yield
     runtime = datetime.now() - start
 
-    asyncio.ensure_future(
-        track(
-            TRAINING_COMPLETED_EVENT,
-            {
-                "training_id": training_id,
-                "model_type": model_type,
-                "runtime": int(runtime.total_seconds()),
-            },
-        )
+    _track(
+        TRAINING_COMPLETED_EVENT,
+        {
+            "training_id": training_id,
+            "type": model_type,
+            "runtime": int(runtime.total_seconds()),
+        },
     )
 
 
 @ensure_telemetry_enabled
-async def track_telemetry_disabled() -> None:
+def track_telemetry_disabled() -> None:
     """Track when a user disables telemetry."""
+    _track(TELEMETRY_DISABLED_EVENT)
+
+
+@ensure_telemetry_enabled
+def track_data_split(fraction: float, data_type: Text) -> None:
+    """Track when a user splits data.
+
+    Args:
+        fraction: How much data goes into train and how much goes into test
+        data_type: Is this core, nlu or nlg data
+    """
+    _track(TELEMETRY_DATA_SPLIT_EVENT, {"fraction": fraction, "type": data_type})
+
+
+@ensure_telemetry_enabled
+def track_validate_files(validation_success: bool) -> None:
+    """Track when a user validates data files.
+
+    Args:
+        validation_success: Whether the validation was successful
+    """
+    _track(TELEMETRY_DATA_VALIDATED_EVENT, {"validation_success": validation_success})
+
+
+@ensure_telemetry_enabled
+def track_data_convert(output_format: Text, data_type: Text) -> None:
+    """Track when a user converts data.
+
+    Args:
+        output_format: Target format for the converter
+        data_type: Is this core, nlu or nlg data
+    """
+    _track(
+        TELEMETRY_DATA_CONVERTED_EVENT,
+        {"output_format": output_format, "type": data_type},
+    )
+
+
+@ensure_telemetry_enabled
+def track_tracker_export(
+    number_of_exported_events: int,
+    tracker_store: "TrackerStore",
+    event_broker: Optional["EventBroker"],
```

Comment:
optional as out of the scope of the PR: the type annotation for `_prepare_event_broker` is wrong as `EventBroker` can be `Optional` at this point. However, after `_prepare_event_broker` there is no way the broker can be `None` as this would have led to a nullpointer exception.

Reply:
fixed it :+1: